From cbc79df9286c6f67af371fce221737bd1f7677c1 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Tue, 09 Mar 2021 06:27:51 +0200
Subject: [PATCH] [SW-0] bug fix in TS unit reset flow

writing to 1 UNIT_RST_N is not enough, need to write 0 into
UNIT_RST_N_L register

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: If88a45d598228060416a6e150e5dcdd58bebf3ee
---

diff --git a/preboot/thermal_v2.c b/preboot/thermal_v2.c
index a24228e..2069b62 100644
--- a/preboot/thermal_v2.c
+++ b/preboot/thermal_v2.c
@@ -263,5 +263,6 @@
 	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_H);
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
 	udelay(1000);
-	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
+	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_L);
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
 }
